extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php

```diff
@@ -2,6 +2,59 @@
 
 class PremiumPageHeaderController extends WikiaController {
 
+	public function articleHeader() {
+		$skinVars = $this->app->getSkinTemplateObj()->data;
+		$this->content_actions = $skinVars['content_actions'];
+
+		// edit button preparation
+		// action button (edit / view soruce) and dropdown for it
+		$this->prepareActionButton();
+		// dropdown actions
+		$this->dropdown = $this->getDropdownActions();
+
+		// comments/talk button
+		$commentsEnabled = $this->checkArticleComments();
+		$this->commentButtonMsg = $commentsEnabled ? 'oasis-page-header-comments' : 'oasis-page-header-talk';
+		$this->commentsLink = $this->getCommentsLink();
+
+		$this->setVal( 'displaytitle', $skinVars['displaytitle'] );
+		$this->setVal( 'title', $skinVars['title'] );
+
+		$categoryLinks = $this->getContext()->getOutput()->getCategoryLinks();
+		$normalCategoryLinks = $categoryLinks['normal'] ?? [];
+
+		$visibleCategoriesLimit = 4;
+		if ( count( $normalCategoryLinks ) > 4 ) {
+			$visibleCategoriesLimit = 3;
+		}
+		$categories = array_slice( $normalCategoryLinks, 0, $visibleCategoriesLimit );
+		$visibleCategories = $this->extendWithTrackingAttribute( $categories, 'categories' );
+		$extendedCategories = array_slice( $normalCategoryLinks, $visibleCategoriesLimit );
+		$moreCategories = $this->extendWithTrackingAttribute( $extendedCategories, 'categories-more' );
+
+		$this->setVal( 'inCategoriesText', wfMessage( 'pph-in-categories' )->plain() );
+		$this->setVal( 'visibleCategories', $visibleCategories );
+		$this->setVal( 'moreCategoriesText', wfMessage( 'pph-categories-more' )->numParams( count( $moreCategories ) )->text() );
+		$this->setVal( 'moreCategoriesLength', count( $moreCategories ) );
+		$this->setVal( 'moreCategories', $moreCategories );
+		$this->setVal( 'curatedContentButton', $this->getEditMainPage() );
+
+		if ( $this->app->getSkinTemplateObj() ) {
```

Comment:
can we move this block to `widgetLanguages()` method and return list of languages from this method?
then we have here only `$this->setVal('language_list', $this->widgetLanguages());`